test/abstract_client.js

```diff
@@ -2983,7 +3019,12 @@ module.exports = function (server, config) {
 
         client.on('connect', function () {
           if (!reconnect) {
-            client.publish('topic', 'payload', { qos: 2 })
+            client.publish('topic', 'payload', { qos: 2 }, function (err) {
+              server2.close()
```

Comment:
Does server.close expects a callback too?

Reply:
It looks like it does. Do you think I should update the tests to use it? I assume it will help, but I can't guarantee that it won't cause other problems.

Reply:
Looks better now :)